server/src/main/java/io/crate/analyze/SwapTableAnalyzer.java

```diff
@@ -68,10 +70,16 @@ public AnalyzedSwapTable analyze(SwapTable<Expression> swapTable,
         }
         SearchPath searchPath = txnCtx.sessionSettings().searchPath();
         Role user = txnCtx.sessionSettings().sessionUser();
-        return new AnalyzedSwapTable(
-            schemas.findRelation(swapTable.source(), Operation.ALTER, user, searchPath),
-            schemas.findRelation(swapTable.target(), Operation.ALTER, user, searchPath),
-            dropSource
-        );
+
+        var source = schemas.findRelation(swapTable.source(), Operation.ALTER_TABLE_RENAME, user, searchPath);
+        if (!(source instanceof DocTableInfo sourceRelation)) {
+            throw new OperationOnInaccessibleRelationException(source.ident(), "Cannot execute SWAP TABLE on views, " + source.ident().name());
+        }
+        var target = schemas.findRelation(swapTable.target(), Operation.ALTER_TABLE_RENAME, user, searchPath);
+        if (!(target instanceof DocTableInfo targetRelation)) {
+            throw new OperationOnInaccessibleRelationException(target.ident(), "Cannot execute SWAP TABLE on views, " + target.ident().name());
+        }
```

Comment:
Currently views are not allowed to be swapped, changing `swap table`'s operation from `Operation.ALTER` to `Operation.ALTER_TABLE_RENAME` allows this. So these extra checks are now needed.